### PR TITLE
Use ApiCompatTask in ApiCompat.proj

### PR DESCRIPTION
**Pull request by ViktorHofer**
Requires https://github.com/dotnet/arcade/pull/8523

As ApiCompat now has a task that allows to run in-proc, use that instead
of shelving out. Also use items instead of properties to define
the attribute exclusion list.

Unfortunately msbuild doesn't support running targets in parallel, only projects. Moving every ApiCompatTask invocation into a separate project is overkill and just increases the overall evaluation and restore time. Therefore stick with running the ApiCompat invocations sequentially.

### CODE REVIEW

File: src/libraries/apicompat/ApiCompat.proj [hunk 1]
```diff
@@ -8,14 +8,15 @@
     <CompileUsingReferenceAssemblies>true</CompileUsingReferenceAssemblies>
     <TrimOutPrivateAssembliesFromReferencePath>true</TrimOutPrivateAssembliesFromReferencePath>
 
-    <ApiCompatNSBaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netcoreapp.netstandard.txt</ApiCompatNSBaselineFile>
-    <ApiCompatNSOnlyBaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netcoreapp.netstandardOnly.txt</ApiCompatNSOnlyBaselineFile>
-    <PreviousNetCoreAppBaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.PreviousNetCoreApp.txt</PreviousNetCoreAppBaselineFile>
+    <ApiCompatNetStandard20BaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netcoreapp.netstandard.txt</ApiCompatNetStandard20BaselineFile>
+    <ApiCompatNetStandard21BaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netcoreapp.netstandardOnly.txt</ApiCompatNetStandard21BaselineFile>
```
Comment (bartonjs) [inline src/libraries/apicompat/ApiCompat.proj:12]: I don't actually know what the "netstandardOnly" file means; but if it's something about .NET Standard 2.1 it'd be nice if the filename was also changed to reflect that.  Perhaps as a followup.

File: src/libraries/apicompat/ApiCompat.proj [hunk 2]
```diff
@@ -35,60 +36,50 @@
...
-      <ApiCompatExitCode>0</ApiCompatExitCode>
-      <!-- TODO: Replace with ReferencePathWithRefAssemblies when ApiCompat supports passing in a list of assemblies: https://github.com/dotnet/arcade/issues/8394. -->
-      <ApiCompatArgs>--impl-dirs "$(LibrariesAllRefArtifactsPath.TrimEnd('\/'))"</ApiCompatArgs>
-      <ApiCompatArgs Condition="'@(ApiCompatExcludeAttributeFile)' != ''">$(ApiCompatArgs) --exclude-attributes "@(ApiCompatExcludeAttributeFile->Metadata('FullPath'), ',')"</ApiCompatArgs>
-    </PropertyGroup>
-
-    <WriteLinesToFile File="$(ApiCompatResponseFile)"
-                      Lines="$(ApiCompatArgs)"
-                      Overwrite="true" />
+    <Message Text="ApiCompat -> Comparing $(NetCoreAppCurrent) reference assemblies against .NETStandard2.x and .NETCoreApp$(NetCoreAppLatestStableVersion)..." Importance="high"  />
```
Comment (bartonjs) [inline src/libraries/apicompat/ApiCompat.proj:41]: I do like when something is clear about what failed... but do we have precedent for using "High"?

Comment (bartonjs) [reply src/libraries/apicompat/ApiCompat.proj:41]: Nevermind, we used it previously, but text and importance changed places.

File: src/libraries/apicompat/ApiCompat.proj [hunk 3]
```diff
@@ -35,60 +36,50 @@
...
 
-    <!-- In order to update the .NETStandard baseline, you can just start the build with /p:UpdateNETStandardBaseline=true -->
-    <PropertyGroup>
-      <NetStandardLibrary20RefPath>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', 'netstandard.library', '$(NetStandardLibraryVersion)', 'build', 'netstandard2.0', 'ref'))</NetStandardLibrary20RefPath>
-      <NetStandard21OnlyRef>$(NETStandard21RefPath)netstandard.dll</NetStandard21OnlyRef>
-      <NetStandard21BaselineModifer>--baseline</NetStandard21BaselineModifer>
-      <NetStandard21BaselineModifer Condition="'$(UpdateNETStandardBaseline)' == 'true'">&gt;</NetStandard21BaselineModifer>
```
Comment (bartonjs) [inline src/libraries/apicompat/ApiCompat.proj:57]: The three comparisons used to use three different update flags, now they share one.

Do all three tasks run and report errors before terminating?  If not, it feels surprising that you get an error on the first one, run the update command and "boom" three files got edited.  ("But where'd the other two come from? Only one had errors...").

Not saying I disagree with collapsing them, just that it seems potentially surprising.

Comment (ViktorHofer) [reply src/libraries/apicompat/ApiCompat.proj:57]: The baseline files for netstandard2.0 and netstandard2.1 haven't been updated for years. Updating them all-together makes it less likely that some of them get stale. Note that there's an ApiCompat switch that removes baseline entries that don't apply anymore and we should probably leverage it in a follow-up command. Apart from that, the .NETStandard baseline files usually don't change much. I expect the "latest stable .NETCoreApp" baseline file to be way noisier.

File: src/libraries/apicompat/ApiCompat.proj (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/apicompat/ApiCompat.proj]: Updated the baseline names.

File: NuGet.config [hunk 4]
```diff
@@ -12,6 +12,7 @@
       one as a template. The following line is a marker to insert the test restore sources.
     -->
     <!-- TEST_RESTORE_SOURCES_INSERTION_LINE -->
+    <add key="vihofer-staging" value="https://www.myget.org/F/vihofer-staging/api/v3/index.json" />
```
Comment (ViktorHofer) [inline NuGet.config:15]: Note that the second commit in this PR which updates this file and the Versions.props file will be reverted before merging. This is just temporary to consume the arcade change and test it out in CI.

File: src/libraries/apicompat/ApiCompatBaseline.netstandard2.1.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
+Compat issues with assembly netstandard:
```
[the remark this answers is not in the document]
Comment (ericstj) [reply src/libraries/apicompat/ApiCompatBaseline.netstandard2.1.txt:1]: Even that view of the diff seems to be missing some single line moves.  I wonder what's causing moves -- did anything change about how its run that might change this order?

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/libraries/apicompat/ApiCompatBaseline.netstandard2.1.txt:1]: I get the exactly same diff on main. Whatever changed the ordering is unrelated to the changes in https://github.com/dotnet/arcade/pull/8523 and this PR.

**Comment by ViktorHofer** [comment]
https://github.com/dotnet/arcade/pull/8523 is now merged and part of this PR. @bartonjs @ericstj please review and approve.

**Comment by ViktorHofer** [comment]
You can see that https://github.com/dotnet/runtime/pull/66504 doesn't include an Arcade update. There won't be any new Arcade dependency updates until this change is in as the ApiCompat change in Arcade requires reaction (via this PR). Because of that it would be great if you could review this trivial change. I verified that the baseline diff is correct.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**